openedx_events/apps.py

```diff
@@ -34,47 +35,65 @@ class OpenedxEventsConfig(AppConfig):
 
     name = "openedx_events"
 
-    def _get_validated_signal_config(self, event_type, configurations):
+    def _get_validated_signal_config(self, event_type, configuration):
         """
         Validate signal configuration format.
 
+        Example expected signal configuration:
+        {
+            "topic_a": { "event_key_field": "my.key.field", "enabled": True },
+            "topic_b": { "event_key_field": "my.key.field", "enabled": False }
+        }
+
         Raises:
             ProducerConfigurationError: If configuration is not valid.
         """
-        if not isinstance(configurations, list) and not isinstance(configurations, tuple):
+        if not isinstance(configuration, dict) and not isinstance(configuration, dict):
             raise ProducerConfigurationError(
                 event_type=event_type,
-                message="Configuration for event_types should be a list or a tuple of dictionaries"
+                message="Configuration for event_types should be a dict"
             )
         try:
             signal = OpenEdxPublicSignal.get_signal_by_type(event_type)
         except KeyError as exc:
             raise ProducerConfigurationError(message=f"No OpenEdxPublicSignal of type: '{event_type}'.") from exc
-        for configuration in configurations:
-            if not isinstance(configuration, dict):
+        for _, topic_configuration in configuration.items():
+            print(f"{topic_configuration=}")
+            if not isinstance(topic_configuration, dict):
                 raise ProducerConfigurationError(
                     event_type=event_type,
-                    message="One of the configuration object is not a dictionary"
+                    message="One of the configuration objects is not a dictionary"
```

Comment:
Imagine everything is working fine, and someone adds a new bad config detail. Do we want that to break all working publishing? Seems scary. Something to consider.

Reply:
Good point. I'm going to punt on this since it requires a little more thought than I want to put it while trying to land this but it will be a fast follow.

Reply:
[punt]

Reply:
Isn't it better to fail early rather than allowing a back config detail? This error will stop the application from starting.